src/amp-story-player/amp-story-player-iframe-pool.js

```diff
@@ -77,4 +77,58 @@ export class IframePool {
 
     return detachedStoryIdx;
   }
+
+  /**
+   * Clears stored story indices.
+   * @return {!Array<number>}
+   */
+  evictStories() {
+    const evictedStories = this.storyIdsWithIframe_;
+    this.storyIdsWithIframe_ = [];
+    return evictedStories;
+  }
+
+  /**
+   * @return {!Array<number>}
+   */
+  getAvailableIframeIdx() {
+    return this.iframePool_;
+  }
+
+  /**
+   * Finds adjacent iframe indices given an index.
+   * Examples (x is provided index):
+   * [x] [1] [2] [] []
+   * [2] [x] [1] [] []
+   * [] [] [2] [1] [x]
+   * [1] [x]
+   * @param {!Array<number>} queue
+   * @param {!Array<number>} adjacent
+   * @param {number} numStories
+   */
+  findAdjacent(queue, adjacent, numStories) {
```

Comment:
This method's arguments are very confusing:
- queue: what is it?
- adjacent: why is it a parameter to provide if it's what we're trying to get?
- numStories: what is it?
- It doesn't actually return anything?

From your JSDoc, we should provide an array of item and an index?

Can we just build and return the array of adjacent stories, without accepting an `adjacent` array?
If you need it to use recursive code you can use a closure instead.

Reply:
Done.